File: MetInProximityBack/MetInProximityBack/Providers/MicrosoftOAuthProvider.cs
using MetInProximityBack.Interfaces;
using MetInProximityBack.NewFolder;
using MetInProximityBack.Types.OAuth;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Configuration;
using System.Security.Claims;
using static Microsoft.EntityFrameworkCore.DbLoggerCategory;

namespace MetInProximityBack.Providers
{
    public class MicrosoftOAuthProvider : IOAuthProvider
    {
        public string ProviderName => "microsoft";

        public OAuthUserDto MapResponseToUser(IEnumerable<Claim> res)
        {
            var userEmail = res.GetClaimValue("email");
            var userName = res.GetClaimValue("name");

            return new OAuthUserDto
            {
                UserName = userName,
                UserEmail = userEmail,
                IsEmailVerified = userEmail == null ? false : true,
            };

        }

    }
}


/*
        [Obsolete("Function fufilled on client side")]
        public string TokenUrl => "https://login.microsoftonline.com/common/oauth2/v2.0/token";
        [Obsolete("Function fufilled on client side")]
        public string UserUrl => "https://graph.microsoft.com/v1.0/me";

        public MicrosoftOAuthProvider(
            IConfiguration configuration
        )
        {
            _configuration = configuration;
        }

        [Obsolete("Function fufilled on client side")]
        public Dictionary<string, string> GetReqValues(string authCode, string codeVerifier)
        {
            var req = new Dictionary<string, string>
            {
                { "client_id", _configuration["Auth:Microsoft:ClientId"] },
                { "client_secret", _configuration["Auth:Microsoft:ClientSecret"] },
                { "grant_type", "authorization_code" },
                { "code", authCode },
                { "response_mode", "query" },
                { "redirect_uri", _configuration["Auth:Microsoft:RedirectUri"] },
                { "scope", "openid profile email" }
            };

            return req;
        }
        */

File: MetInProximityFront/app/src/main/java/com/example/metinproximityfront/Providers/OAuth/MicrosoftOAuthProvider.kt
package com.example.metinproximityfront.Providers.OAuth

import com.example.metinproximityfront.Interfaces.OAuthProvider

class MicrosoftOAuthProvider : OAuthProvider {
    override val name = "Microsoft"
    override val oauthUrl = "https://login.microsoftonline.com/common/oauth2/v2.0/authorize"
    override val redirectUri = "https://localhost:7238/api/account/oauth/microsoft"
    override val clientId = ""

    override fun getOAuthUrl(): String {
        // You can customize the URL with parameters like client ID, scopes, etc.
        return oauthUrl +
                "?client_id=$clientId" +
                "&redirect_uri=$redirectUri" +
                "&response_type=code" +
                "&response_mode=query" +
                "&scope=openid%20profile%20email"
    }

}